### PR TITLE
Add view-component for input fields

### LINKED ISSUE #41: Two view components could be merged into one
It appears to me that project_parameters and scenario_parameters view-component could be described by a single more generic view component. Both of them define input parameters. I therefore think that the view-component could describe what are the features we want to have for generic input parameters such as:

- Each parameter can have explanatory text with a ? icon which describes briefly the parameter and provide a link to documentation for in depth explanation (a link to this icon file would be an attribute of the view-component "input-parameters")

- Each parameter has a type (scalar, str, timeseries, matrix, array) and we could describe how the user can input things that are not scalar or str (e.g. a "..." icon next to the parameter name instead of a text input for timeseries)

- If the user changes a parameter's value, then something should happen to signify the user that what is displayed on the screen is not yet what is simulated (a parameter can been seen as changed on the GUI, however the results can still be the previous one, we need to prevent this type of confusion to happen)

On of the attribute would be "List of parameters", then the scenario_parameters view-component could simply be expressed as this "input parameters" view-component with a list of parameters.

For this we probably need an abstract definition of what a "parameter" is and which attributes it has as well (right now I see type, unit, value)

### PR DESCRIPTION
**Pull request by mahendrark**
Fix #41 

**Changes proposed in this pull request**:
- Add an abstract view-component that describes the input fields in the UI

The following steps were realized, as well (if applies):
- [x] Update the CHANGELOG.md

<sub>*For more information on how to contribute check the [CONTRIBUTING.md](https://github.com/rl
-institut/open_plan/blob/dev/CONTRIBUTING.md).*<sub>

### CODE REVIEW

File: docs/abstract_components/input_parameter_field.rst [hunk 1]
```diff
@@ -0,0 +1,38 @@
+Input Parameter Field
+---------------------
+
+This view-component enables the user input appropriate data in different views.
+
+Attributes
+^^^^^^^^^^
+
+**Input Field Label**
```
Comment (Bachibouzouk) [inline docs/abstract_components/input_parameter_field.rst]:
```suggestion
**Label**
```

**Comment by mahendrark** [comment]
@Bachibouzouk 

This view-component links to many, many views. In this case, what should be under the "Links to other views" ?

File: docs/tool_interface.rst [hunk 2]
```diff
@@ -156,4 +156,8 @@ Abstract-components Definition
 
 .. include:: abstract_components/scenario.rst
 
+.. _input-field-label:
```
Comment (Bachibouzouk) [inline docs/tool_interface.rst]:
```suggestion
.. _input_parameter_field-label:
```

see the view_component_template.rst

**Comment by Bachibouzouk** [comment]
> @Bachibouzouk
> 
> This view-component links to many, many views. In this case, what should be under the "Links to other views" ?

I would only link it to the view-components, not to the view containing the view-components

**Comment by Bachibouzouk** [comment]
This input_parameter_field could be a view_component to me, otherwise we need to define a view component which has exactly the same attributes, only rendering in extra. That would be duplicate

**Comment by mahendrark** [comment]
> This input_parameter_field could be a view_component to me, otherwise we need to define a view component which has exactly the same attributes, only rendering in extra. That would be duplicate

I agree. I will move it to view_components then.

File: docs/tool_interface.rst [hunk 3]
```diff
@@ -157,10 +157,15 @@ View-components definition
```
Comment (Bachibouzouk) [inline docs/tool_interface.rst]: I already rebased locally and add my changes, do not push more commits, they will be overwritten

**Review by Bachibouzouk — APPROVED** [review]

**Pull request merged**